libs/settings/api/src/lib/settings.resolver.spec.ts

```diff
@@ -0,0 +1,151 @@
+import {Test, TestingModule} from '@nestjs/testing'
+import {INestApplication, Module} from '@nestjs/common'
+import request from 'supertest'
+import * as crypto from 'crypto'
+import {GraphQLModule} from '@nestjs/graphql'
+import {PrismaClient} from '@prisma/client'
+import {ApolloDriverConfig, ApolloDriver} from '@nestjs/apollo'
+import {PrismaModule} from '@wepublish/nest-modules'
+import {SettingsResolver} from './settings.resolver'
+import {SettingsService} from './settings.service'
+import {GraphQLSettingValueType} from './settings.model'
+import {SettingName} from './setting'
+
+export const generateRandomString = () => crypto.randomBytes(20).toString('hex')
+
+@Module({
+  imports: [
+    GraphQLModule.forRoot<ApolloDriverConfig>({
+      driver: ApolloDriver,
+      autoSchemaFile: true,
+      path: '/',
+      cache: 'bounded'
+    }),
+    PrismaModule
+  ],
+  providers: [SettingsResolver, SettingsService, GraphQLSettingValueType]
+})
+export class AppModule {}
+
+const settingsListQuery = `
+  query settingsList($filter: SettingFilter) {
+    settingsList(filter: $filter) {
+      id
+      name
+      value
+      settingRestriction {
+        maxValue
+        minValue
+        inputLength
+        allowedValues {
+          stringChoice
+          boolChoice
+        }
+      }
+    }
+  }
+`
+
+const settingQuery = `
+  query setting($id: String!) {
+    setting(id: $id) {
+      id
+      name
+      value
+      settingRestriction {
+        maxValue
+        minValue
+        inputLength
+        allowedValues {
+          stringChoice
+          boolChoice
+        }
+      }
+    }
+  }
+`
+
+const updateSettingsMutation = `
+  mutation updateSettings($value: [UpdateSettingInput!]!) {
+    updateSettings(value: $value) {
+      value
+    }
+  }
+`
+
+describe('SettingsResolver', () => {
+  let app: INestApplication
+  let prisma: PrismaClient
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [AppModule]
+    }).compile()
+
+    prisma = module.get<PrismaClient>(PrismaClient)
```

Comment:
I think we said we wanted to mock the DB (unlike what we have now), right @tomaszdurka ?